src/contrasts.jl

```diff
@@ -101,19 +101,19 @@ of Memory and Language, 110_, 104038. https://doi.org/10.1016/j.jml.2019.104038
 abstract type AbstractContrasts end
 
 # Contrasts + Levels (usually from data) = ContrastsMatrix
-struct ContrastsMatrix{C <: AbstractContrasts, T, U}
-    matrix::Matrix{Float64}
+struct ContrastsMatrix{C <: AbstractContrasts, M <: AbstractMatrix, T, U}
+    matrix::M
```

Comment:
Any particular reason to put `M` in second position? Putting it at the end would avoid breakage.

Reply:
my thinking here (near as I can reconstruct it at this point) was that we're already doing a breaking release, and this position puts it in a place where you can specify it without needing to specify the other parameters.

Reply:
OK so you suppose it will be more common to specify `M` than `T` or `U`?

Reply:
yeah definitely.  those are only there to make the contrast lookup type stable/inferrable